### ISSUE TITLE
Allow bitrates higher than 6000kbps (way higher).

### ISSUE THREAD

**Issue by hluengas** [owncast/owncast#1462]
Owncast is awesome because its the only easy way for me to stream gamplay at over 60FPS to my friends. However 6Mbps isn't even close to a high enough bitrate for 90FPS 1440p or 4K streaming even with the highest quality CPU ussage. Why not make the bitrate slider cap out at something really high-end like 50Mbps (or if you think people will have a bad experience when they set that without actually having the bandwidth, then at least provide a manual entry method). I tried to change this myself and build from source, but I believe this is controlled by the owncast-admin side of things and I can't figure out how to pull in changes to that using the provided Dockerfile.

**Comment by gabek** [comment]
You're right, the reason it's not a freeform text box, or a slider that goes to crazy values is because people will put in crazy values.

Owncast aims to be a sane solution for people looking to self-host their live streams. Most of these people self-hosting won't have a super server or infinite bandwidth. Sure, some will, like you.  But instead what will happen is people will *think* they have a super server with infinite bandwidth, everything will go horribly, and they'll delete Owncast and tell everyone how it sucks. We already see this now with other values. Because we offer it people believe it will work for them.

I suggest using the APIs to set whatever crazy values you want, even though they're not supported in the admin. https://owncast.online/api/latest/#tag/Admin/paths/~1api~1admin~1config~1video~1streamoutputvariants/post

If this is something that becomes a normal use case we can think of some special, hidden place to go and type in whatever configuration you want, but as it stands any setting that's "use at your own risk" doesn't work. The risk is ours by offering it.

**Comment by ma-karai** [comment]
I understand that the unique selling point of owncast is that you can run this even on a raspberry pi, and its great.  However, why not cater to people who have high-end infrastructures. You could easily create an option/checkmark, with a question if this is hosted on a high bandwidth infrastructure and then have the slider expand to 20000kbits or whatever.  It does not cost you anything, right ?

Since you linked the api do you have an example curl script to create a stream profile that creates a 12k kbit profile ?

**Comment by gabek** [comment]
It's about guiding the most number of people to the best possible scenarios, and there's multiple things at play for these scenarios.

It's not just about your infrastructure, it's about your viewers. That's the most important part of any of this. If your viewers can't view the stream, then what's the point?

If you max our your infrastructure streaming a 8000k stream, making it so you can't add additional lower bitrate options, only those with the fastest connections who have a geographically optimal connection (not to mention mobile viewers would be out of the question) to your a server will even be able to watch it, leading everyone else to just get a infinite loading spinner and blaming Owncast for not working.

The downside to any "Are you sure you want to do X?" configuration option is people always say "yes". And then when everything doesn't work they blame Owncast for not working. This is very common.

And like the thread above said, you can set any video settings you want, but you have to use the API. Then you're doing your own thing and you're making your own choices. But when it comes to the UI, people will just flip on all the switches, drag all the sliders to the max, and then say "it doesn't work, nobody can watch it, Owncast sucks." I'm doing my best from giving people enough rope to hang themselves with.

**Comment by hluengas** [comment]
> I understand that the unique selling point of owncast is that you can run this even on a raspberry pi, and its great. However, why not cater to people who have high-end infrastructures. You could easily create an option/checkmark, with a question if this is hosted on a high bandwidth infrastructure and then have the slider expand to 20000kbits or whatever. It does not cost you anything, right ?
> 
> Since you linked the api do you have an example curl script to create a stream profile that creates a 12k kbit profile ?

I originally asked this question and while I do still wish they would find someway to incorporate custom settings into the UI, at least the API is actually really simple and straight forward to use. They have documentation on their site and you can see my scripts as an example here:
https://github.com/hluengas/owncast_scripts

<blockquote><img src="https://opengraph.githubassets.com/924b0964546ea0416d0dec258269da1e4da6138823107da92681fe000190a507/hluengas/owncast_scripts" width="48" align="right"><div><img src="https://github.githubassets.com/favicons/favicon.svg" height="14"> GitHub</div><div><strong><a href="https://github.com/hluengas/owncast_scripts">GitHub - hluengas/owncast_scripts: owncast api scripts</a></strong></div><div>owncast api scripts. Contribute to hluengas/owncast_scripts development by creating an account on GitHub.</div></blockquote>

**Comment by ma-karai** [comment]
> .... stuff .....

Firstly, owncast is amazing, and you did an awesome job.


I completely understand where you are coming from, however, if my "audience" is only a selected few, why limit me. If someone wants to stream for a random, unknown amount of people with dial-up modems, they can use the lower settings as well.

It's alright tho, glad someone created some scripts. Thanks, Henry @hluengas

**Comment by naboo** [comment]
@hluengas great script. I was also trying to find if/how OwnCast could support higher streams like 4k and found this issue. Do you have any recommendation or experience on hardware (and values) needed to support 4k streaming?

I'm thinking of off-loading the actual streaming to S3 but for the incoming stream and compiling.

**Comment by gabek** [comment]
> Do you have any recommendation or experience on hardware (and values) needed to support 4k streaming?

This document may be helpful to you. https://owncast.online/docs/resources-requirements/


<blockquote><img src="/images/owncast-logo-1000x1000.png" width="48" align="right"><div><img src="https://owncast.online/favicon.png" height="14"> Owncast</div><div><strong><a href="/docs/resources-requirements/">Resources and requirements</a></strong></div><div>There is no hard and fast rule for how much resources Owncast will use, since it depends on your configuration and requirements, but here are some examples.</div></blockquote>

**Comment by realies** [comment]
I want to increase the limit to at least 20000kbits too.
Happy with a warning after 6000kbits or whatever you think 'people should use'.

**Comment by RedQueenAI** [comment]
The 6000 Kbps cap is causing issues with my stream quality. I need at least 12000 Kbps to stream in 4K at 60 FPS without problems. At 6000 Kbps, the stream is buggy and the quality is poor. When I stream at 12000 Kbps, the issues are resolved and the quality is much better.

I use a server and a restreamer to avoid the 6000 Kbps limit, but it's still a challenge. Owncast is difficult to use because it also has a 6000 Kbps cap. This limit is fine for 1080p streaming, but it's not sufficient for 4K streaming.

We have the necessary resources, including a Xeon CPU and a gigabit fiber network connection, but we're restricted by this maximum setting.

**Comment by gabek** [comment]
@Catflower75 Hi there. You can set any bitrate you like through the API. Read the thread above, and you can use your browser's dev tools to customize it to your requirements.

**Comment by RedQueenAI** [comment]
no worked i tryed do dev tool put slider to 12000 even aply it say is 6000.... so i have do a other way to do since need edit source code for can change it never sucess other way slider or other no worked for me so that is for 4k at 60fps for me no a crazy value for 4k since 6000 kps at 60fps 4k good luck full buged as quality issue and other sure 1080p 60 fps 6000 ok no for 4k... like i do	StreamVariants: []models.StreamOutputVariant{
			{
				IsAudioPassthrough: true,
				VideoBitrate:       12000,
				Framerate:          60,
				CPUUsageLevel:      2,
			},
		},

export const VIDEO_BITRATE_DEFAULTS = {
  min: 400,
  max: 12000,
  defaultValue: 12000,
  unit: 'kbps',
  incrementBy: 100,
  tip: 'The overall quality of your stream is generally impacted most by bitrate.',
};

export const DEFAULT_VARIANT_STATE: VideoVariant = {
  framerate: 60,
  videoPassthrough: false,
  videoBitrate: 12000,
  audioPassthrough: true, // if false, then CAN set audiobitrate
  audioBitrate: 0,
  cpuUsageLevel: 2,
  scaledHeight: null,
  scaledWidth: null,
  name: '',
};

**Comment by gabek** [comment]
That's really not necessary. Like I said above, you can use the API. But I'm glad you found your own solution!

**Comment by kotx** [comment]
> ...instead what will happen is people will _think_ they have a super server with infinite bandwidth, everything will go horribly, and they'll delete Owncast and tell everyone how it sucks. We already see this now with other values. Because we offer it people believe it will work for them.

Why not put a warning in the UI for this, hide it behind a checkbox, and recommend a sane value as the maximum? I don't see why people have to use hacky workarounds (especially if they aren't developers/technical people) for their _valid_ use cases. (perhaps an "advanced" section of the settings would work too)
I would think hiding functionality behind workarounds would make people "delete Owncast and tell everyone how it sucks" too.

**Comment by gabek** [comment]
> Why not put a warning in the UI for this, hide it behind a checkbox

That's what we did with video passthrough, and learned that everyone turned it on anyway. It's still the number one cause of issues when people report problems. Making people click through to "advanced" or agree that "this setting should not be used for most people" doesn't stop them from using it. It will result in problems, and then they'll report that Owncast doesn't work, and then I need to troubleshoot with them.

If there was a **good** way to do advanced settings where truly only **advanced** people will use them, then we could do things like you suggested. But until that is figured out, we have to optimize for the 95% of people, not for the 5% of people.

**Comment by gabek** [comment]
Since it hasn't been clear what using the API looks like, here's a simple example that literally anybody can do to make any kind of changes to their video configuration that they like. Keep in mind this is unsupported, but it's your server and you can do anything you like with it.

![capture](https://github.com/owncast/owncast/assets/414923/070d6e30-5498-476c-a6a2-7e0bb7c58d81)

**Comment by JeremyMahieu** [comment]
Imagine having to explain to users to go edit the request in dev console instead of providing a reasonable value in the UI. Why are you so against this? 15000kbps isn't that high.

**Comment by realies** [comment]
> Imagine having to explain to users to go edit the request in dev console instead of providing a reasonable value in the UI. Why are you so against this? 15000kbps isn't that high.

Some devs are just unreasonably stubborn. It sounds like they fear facing more support requests if higher bitrate is easily configurable.

However, that logic is flawed, because if the motivation is struggle or annoyance with the amount of support requests, they should limit the bitrate further until nobody uses the app and the issues section becomes inactive.

Another theory is blind faith in what Twitch has decided for the max common bitrate, which is probably based on optimising aggregate bandwidth costs and not on issues with clients unable to stream higher bitrates.

**Comment by gabek** [comment]
I actually have no idea what Twitch's settings are. My reasons are documented above. [Like I mentioned](https://github.com/owncast/owncast/issues/1462#issuecomment-2122866367), i'm not against changing it if I had a solution that worked, but I don't currently have it.

**Comment by scarignan** [comment]
> Since it hasn't been clear what using the API looks like, here's a simple example that literally anybody can do to make any kind of changes to their video configuration that they like. Keep in mind this is unsupported, but it's your server and you can do anything you like with it.
> 
> ![capture](https://github.com/owncast/owncast/assets/414923/070d6e30-5498-476c-a6a2-7e0bb7c58d81) [ ![capture](https://github.com/owncast/owncast/assets/414923/070d6e30-5498-476c-a6a2-7e0bb7c58d81) ](https://private-user-images.githubusercontent.com/414923/332976668-070d6e30-5498-476c-a6a2-7e0bb7c58d81.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDI5MzI3NzQsIm5iZiI6MTc0MjkzMjQ3NCwicGF0aCI6Ii80MTQ5MjMvMzMyOTc2NjY4LTA3MGQ2ZTMwLTU0OTgtNDc2Yy1hNmEyLTdlMGJiN2M1OGQ4MS5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMzI1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDMyNVQxOTU0MzRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xMzFiOWQ1MmExZTQ4NDc4MjhkYmMwMGM3YzM5MTNkNjRjNTA4YWViZGY5ZjliOWNhNjAxY2NkY2IyODRlZjI4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.a3Jch1W1FueUDVXNNmqV1u3hhGKyHa20Qdg4u9z8gC4)

Thanks for actually providing a workaround. I agree that it would be great to just unlock the slider with a warning. We are in 2025, most people have a 1gb, download/upload connexion and I do stream in 2k Ultra-wide. 16000kbps/ is the minimum configuration for me to go with so people can have a gaming stream that isnt blurry. I do provide a "medium" setting in 6000/kbps. while its a good idea for people that has no good internet connexion, the stream will definitely be blurry especially if the game has a lot of action on screen (FPS games). 

Its good to have options in my opinion. 

Thanks for taking this into consideration. 

Regards.

**Comment by gabek** [comment]
> Thanks for actually providing a workaround. I agree that it would be great to just unlock the slider with a warning.

We have a warning with Video Passthrough, and people turn that on and then report that their stream doesn't work all the time (no offense), so unfortunately that's not a solution.

**Comment by gabek** [comment]
The problem I'm trying to solve at this point is the fact that a lot of people run Owncast on low powered hardware, so they can only run one output quality, so they'll select the highest possible. That means if we increase what the highest is, even more people won't be able to watch the stream, and they'll blame it on Owncast, not the streamer.

While the admin stream health page tries to help the admin solve these kind of problems, it doesn't help the type of person who says "I can only do one, why wouldn't I want the best?" not realizing that means a large number of people won't be able to even watch their stream. I understand this is also a problem with people who use Video Passthrough.

I know you'll all say "well that's their problem", but the goal of Owncast has always been to give enough rope to not hang themselves with. You can't trust people to not do stupid things, you have to try and help them to make good decisions. And if we increase the max bitrate to 16000, and everyone starts reporting that every Owncast server they go to is buffering, it makes Owncast look bad.

**Comment by kotx** [comment]
Sounds like you should remove Video Passthrough for the same reasoning then :/

**Comment by gabek** [comment]
> Sounds like you should remove Video Passthrough for the same reasoning then :/

I really should. It's harder to remove something that's already in there, though. That's why I try to put a lot of thought about what goes in.

**Comment by scarignan** [comment]
> I know you'll all say "well that's their problem", but the goal of Owncast has always been to give enough rope to not hang themselves with. You can't trust people to not do stupid things, you have to try and help them to make good decisions. And if we increase the max bitrate to 16000, and everyone starts reporting that every Owncast server they go to is buffering, it makes Owncast look bad.

>The problem I'm trying to solve at this point is the fact that a lot of people run Owncast on low powered hardware, so they can only run one output quality, so they'll select the highest possible. That means if we increase what the highest is, even more people won't be able to watch the stream, and they'll blame it on Owncast, not the streamer.

>While the admin stream health page tries to help the admin solve these kind of problems, it doesn't help the type of person who says "I can only do one, why wouldn't I want the best?" not realizing that means a large number of people won't be able to even watch their stream. I understand this is also a problem with people who use Video Passthrough.

I agree with all you said there but of course in my opinion, you have to have a minimum knowledges in IT when it come to self-hosting these kind of things. The wiki is well documented, if people can't read, well. too bad for them.

For me to be able to stream in 2k without any problem, I had to bump up the number of CPU cores (4 sockets / 4 cores) on my Proxmox (Linux instance) for this to work properly in 16000kbps. (So yes, I would not recommend a raspberry pi for this.)

Anyway, I just want to say, thanks for creating this platform. Its an amazing product. Keep up the good work.

Cheers.

**Comment by FlowerShopGuy** [comment]
I just discovered this awesome thing - "owncast". And really was dissapointed that there is only 6000k... Its so "JPEGY" really sucks... I hope that will be support up to 50K bitrate in the next future versions. For now i'm going back to Nginx + Rtmp + Hls with 50Mbs bitrate.

@gabek 
> The problem I'm trying to solve at this point is the fact that a lot of people run Owncast on low powered hardware, so they can only run one output quality, so they'll select the highest possible. That means if we increase what the highest is, even more people won't be able to watch the stream, and they'll blame it on Owncast, not the streamer.

Just make an notice for that or something then...  Like huge Red font "WANRING BITRATE MORE THAN 6000 Kbs CAN MAKE YOUR STREAM UNPLAYABLE, DO THIS ON YOUR OWN RISK and bla bla" :)

@gabek 
> Since it hasn't been clear what using the API looks like, here's a simple example that literally anybody can do to make any kind of changes to their video configuration that they like. Keep in mind this is unsupported, but it's your server and you can do anything you like with it.


Its works!

**Comment by gabek** [comment]
> Just make an notice for that or something then... Like huge Red font "WANRING BIRATE MORE THAN 6000 Kbs CAN MAKE YOUR STEAM UNPLAYABLE, DO THIS ON YOUR OWN RISK and bla bla" :)

I'll reiterate, that's precisely what we do for video passthrough, and people turn that on and then report that Owncast isn't working all the time. That isn't a solution.

**Comment by nico198x** [comment]
for what it's worth, i just started using Owncast, and am disappointed at the 6k limit. it's pretty low in 2025. top limit should be bumped to at least 8k if not 10k.

if you incorporate AV1 support this will also alleviate the need/desire for obscenely high bitrates.

**Comment by gabek** [comment]
When this discussion first started and continued over the years I had a valid concern based on the types of questions and issues people would run into when running Owncast.

It's been five years with Owncast being a project now, and the types of questions and issues are very different than they were when this issue was first opened.

Originally everyone would complain about buffering, and everyone was configuring their server wrong, people were having issues streaming, and it was making the project look bad, and I was trying to make sure the most people had the best experience. I don't know if we have more/better documentation now or people are more comfortable with the concepts, but these types of issues come up far less often.

That makes me feel far more comfortable increasing this value in the UI. So I think we can do that.

**Comment by gabek** [comment]
That being said. It would still be great to find a way to do advanced settings in the UI in a way that actually worked.

**Comment by nico198x** [comment]
that's great to hear, @gabek ! thanks! 

-------

my thoughts from my duplicate thread:

For stream output settings, the current maximum is 6000, which is an excellent setting for 720p60.

The ideal for 1080p60 with x264 is 13000, with anything more having minimal impact.

I'm requesting to up the max limit so we no longer have to do the HTML workaround, as 6000 is a bit limited for those with faster upload speeds.

If you're concerned about users making the settings too high, you already have the appropriate warnings and instruction on the Video Configuration page.

**Comment by gabek** [comment]
> If you're concerned about users making the settings too high, you already have the appropriate warnings and instruction on the Video Configuration page.

That part still hasn't changed. Nobody reads them, and people click through the warnings entirely ignoring them :)
